.github/workflows/appimage.yml

```diff
@@ -72,15 +85,53 @@ jobs:
       run: |
         sudo -u testbot bash -lc 'cd rdk/services/slam/builtin && sudo go test -v -run TestCartographerIntegration'
 
-    - name: Build AppImage
-      run: sudo -u testbot bash -lc 'cd viam-cartographer && make BUILD_CHANNEL="latest" appimage'
-
     - name: Authorize GCP Upload
       uses: google-github-actions/auth@v1
       with:
         credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
 
-    - name: Deploy Files
+    - name: Build AppImage (PR)
+      if: contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests')
+      run: |
+         sudo -u testbot bash -lc 'cd viam-cartographer && make BUILD_CHANNEL="pr-${{ github.event.pull_request.number }}" appimage'
+
+    - name: Upload Files (PR)
+      if: contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests')
+      uses: google-github-actions/upload-cloud-storage@v0.10.4
+      with:
+        headers: "cache-control: no-cache"
+        path: 'viam-cartographer/etc/packaging/appimages/deploy/'
+        destination: 'packages.viam.com/apps/slam-servers/'
+        glob: '*'
+        parent: false
+        gzip: false
+
+    - name: Add AppImage Links (PR)
```

Comment:
[note] Can add AppImage links in one comment but since the x86_64 takes longer there is the possibility that it can be clicked before it has been truly updated (or exists)

Reply:
You should be able to solve this pretty easily. Just have the "add links" bit be it's own job, and depend on the full appimage flow. Then it'll only run when both parts of the matrix are complete.

Or you can have another trigger for completion like we do in RDK (see the code coverage workflow, which also posts the appimage links.)